src/universal/hooks/useTooltip.ts

```diff
@@ -1,45 +1,35 @@
-import {useMemo} from 'react'
-import getBBox, {RectElement} from 'universal/components/RetroReflectPhase/getBBox'
-import useCoords, {MenuPosition, UseCoordsOptions} from 'universal/hooks/useCoords'
+import {useCallback, useRef} from 'react'
+import useCoords, {MenuPosition} from 'universal/hooks/useCoords'
 import useTooltipPortal from 'universal/hooks/useTooltipPortal'
-import usePortal, {PortalStatus, UsePortalOptions} from 'universal/hooks/usePortal'
+import usePortal from 'universal/hooks/usePortal'
 
-interface Options extends UsePortalOptions, UseCoordsOptions {
-  loadingWidth?: number
-}
-
-export interface TooltipProps {
-  openPortal: () => void
-  closePortal: () => void
-  portalStatus: PortalStatus
+interface Options {
+  delay?: number
 }
 
 const useTooltip = (preferredMenuPosition: MenuPosition, options: Options = {}) => {
-  const {onOpen, onClose, originCoords} = options
-  const {targetRef, originRef, coords} = useCoords(preferredMenuPosition, {
-    originCoords
-  })
-  if (originCoords) {
-    (originRef as any).current = {getBoundingClientRect: () => originCoords} as RectElement
-  }
-  const {portal, openPortal, closePortal, togglePortal, portalStatus} = usePortal({
-    onOpen,
-    onClose
-  })
-  const loadingWidth = useMemo(() => {
-    if (options.loadingWidth) return options.loadingWidth
-    const bbox = getBBox(originRef.current)
-    return Math.max(40, bbox ? bbox.width : 40)
-  }, [originRef.current])
-  const tooltipPortal = useTooltipPortal(portal, targetRef, loadingWidth, coords, portalStatus)
-  const tooltipProps = {portalStatus, openPortal, closePortal}
+  const delay = options.delay || 0
+  const {targetRef, originRef, coords} = useCoords(preferredMenuPosition)
+  const {portal, openPortal, closePortal, portalStatus} = usePortal()
+  const tooltipPortal = useTooltipPortal(portal, targetRef, coords, portalStatus)
+  const openDelayRef = useRef<number>()
```

Comment:
@mattkrick why is the type here `number`?

Reply:
return value of setTimeout

Reply:
Refs are no longer just for elements, now they're just dummy objects that are persisted throughout render cycles, so you can keep anything in them. so instead of a class component doing this: `this.openDelayTimeout = setTimeout(() => {})` you have a functional component using a ref

Reply:
I think I get it, thanks for the clarity!